test/parallel/test-single-executable-application.js

```diff
@@ -0,0 +1,135 @@
+
+if (process.config.variables.asan)
+  common.skip('Running the resultant binary fails with `Segmentation fault (core dumped)`.');
+
+if (process.platform === 'linux' && process.config.variables.is_debug === 1)
+  common.skip('Running the resultant binary fails with `Couldn\'t read target executable"`.');
+
+if (process.config.variables.node_shared)
+  common.skip('Running the resultant binary fails with ' +
+    '`/home/iojs/node-tmp/.tmp.2366/sea: error while loading shared libraries: ' +
+    'libnode.so.112: cannot open shared object file: No such file or directory`.');
+
+if (!process.config.variables.node_use_openssl || process.config.variables.node_shared_openssl)
+  common.skip('Running the resultant binary fails with `Node.js is not compiled with OpenSSL crypto support`.');
+
+if (process.config.variables.want_separate_host_toolset !== 0)
+  common.skip('Running the resultant binary fails with `Segmentation fault (core dumped)`.');
+
+if (process.platform === 'linux') {
+  try {
+    const osReleaseText = readFileSync('/etc/os-release', { encoding: 'utf-8' });
+    if (!/^NAME="Ubuntu"/.test(osReleaseText)) {
+      throw new Error('Not Ubuntu.');
+    }
+  } catch {
+    common.skip('Only supported Linux distribution is Ubuntu.');
+  }
+
+  if (process.arch !== 'x64') {
+    common.skip(`Unsupported architecture for Linux - ${process.arch}.`);
+  }
+}
+
+const inputFile = join(tmpdir.path, 'sea.js');
+const requirableFile = join(tmpdir.path, 'requirable.js');
+const outputFile = join(tmpdir.path, process.platform === 'win32' ? 'sea.exe' : 'sea');
+
+tmpdir.refresh();
+
+writeFileSync(requirableFile, `
+module.exports = {
+  hello: 'world',
+};
+`);
+
+writeFileSync(inputFile, `
+const { Module: { createRequire } } = require('module');
+const createdRequire = createRequire(__filename);
+
+// Although, require('../common') works locally, that couldn't be used here
+// because we set NODE_TEST_DIR=/Users/iojs/node-tmp on Jenkins CI.
+const { expectWarning } = createdRequire(${JSON.stringify(join(__dirname, '..', 'common'))});
+
+expectWarning('ExperimentalWarning',
+              'Single executable application is an experimental feature and ' +
+              'might change at any time');
+
+const { deepStrictEqual, strictEqual, throws } = require('assert');
+const { dirname } = require('path');
+
+deepStrictEqual(process.argv, [process.execPath, process.execPath, '-a', '--b=c', 'd']);
+
+strictEqual(require.cache, undefined);
+strictEqual(require.extensions, undefined);
+strictEqual(require.main, module);
+strictEqual(require.resolve, undefined);
+
+strictEqual(__filename, process.execPath);
+strictEqual(__dirname, dirname(process.execPath));
+strictEqual(module.exports, exports);
+
+throws(() => require('./requirable.js'), {
+  code: 'ERR_UNKNOWN_BUILTIN_MODULE',
+});
+
+const requirable = createdRequire('./requirable.js');
+deepStrictEqual(requirable, {
+  hello: 'world',
+});
+
+console.log('Hello, world! 😊');
+`);
```

Comment:
Maybe put this into `test/fixtures/`?

Reply:
Do you know what we can do for this part of this script?
```js
`
...
// Although, require('../common') works locally, that couldn't be used here
// because we set NODE_TEST_DIR=/Users/iojs/node-tmp on Jenkins CI.
const { expectWarning } = createdRequire(${JSON.stringify(join(__dirname, '..', 'common'))});
...
`
```
Especially considering that the `tmpdir.path` has different values locally and on CI (`NODE_TEST_DIR=/Users/iojs/node-tmp` on Jenkins CI)?

Reply:
You could e.g. pass it in through an environment variable when running the resulting executable?